### PR TITLE
Add upstream trigger-jenkins-job Task

### PR DESCRIPTION
**Pull request by rcarrillocruz**

### CODE REVIEW

**Comment by rcarrillocruz** [comment]
@flacatus let me know when you have some time, thx

File: common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml [hunk 1]
```diff
@@ -0,0 +1,120 @@
...
+      description: Jenkins secret containing credentials
+      default: jenkins-credentials
+    - name: JOB_PARAMS
+      type: array
+      description: Extra arguments to append as a part of CURL request
+      default: [""]
+  steps:
+    - name: trigger-pipeline
+      image: registry.access.redhat.com/ubi8/ubi:8.2
+      workingDir: $(workspaces.source.path)
```
Comment (flacatus) [inline common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml:39]: Let's change and add for this tasks volume mounts

File: common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml [hunk 2]
```diff
@@ -0,0 +1,120 @@
+apiVersion: tekton.dev/v1beta1
+kind: Task
+metadata:
+  name: trigger-jenkins-job
+  labels:
+    app.kubernetes.io/version: "0.1"
+  annotations:
```
Comment (flacatus) [inline common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml:7]: we dont need the annotations

File: common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    - name: JENKINS_SECRETS
+      type: string
+      description: Jenkins secret containing credentials
+      default: jenkins-credentials
+    - name: JOB_PARAMS
+      type: array
+      description: Extra arguments to append as a part of CURL request
+      default: [""]
+  volumes:
+    - name: konflux-test-infra-volume
```
Comment (flacatus) [inline common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml:27]: jenkins-credentials-volume

File: common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml [hunk 4]
```diff
@@ -0,0 +1,117 @@
...
+      secret:
+        secretName: konflux-test-infra
```
Comment (flacatus) [inline common/tasks/trigger-jenkins-job/0.1/trigger-jenkins-job.yaml:29]: jenkins-credentials

**Comment by flacatus** [comment]
we are planning to move our tasks under $ROOT/tasks/*. Can you please add this under: 
> tasks/triggers/jenkins/0.1/*.yaml

File: common/tasks/trigger-jenkins-job/0.1/README.md [hunk 5]
```diff
@@ -0,0 +1,3 @@
+# Tekton Task: Trigger Jenkins Job
+
+Please read/update [trigger-jenkins-job tasks docs](../../../../docs/qe-available-tasks/trigger-jenkins-job.md) from docs folder.
```
Comment (flacatus) [inline common/tasks/trigger-jenkins-job/0.1/README.md:3]: Can you add an description to the .md you reference here on how this task is used?. More users that have konflux will benefit from this

**Review by flacatus — CHANGES_REQUESTED** [review]

**Comment by flacatus** [comment]
please add it to: tasks/triggers/jenkins/0.1/jenkins-trigger.yaml

File: tasks/trigger/jenkins/0.1/.yaml [hunk 6]
```diff
@@ -0,0 +1,117 @@
...
+      default: [""]
+  volumes:
+    - name: jenkins-credentials-volume
+      secret:
+        secretName: jenkins-credentials
+  steps:
+    - name: trigger-pipeline
+      image: registry.access.redhat.com/ubi8/ubi:8.2
+      volumeMounts:
+        - name: konflux-test-infra-volume
```
Comment (flacatus) [inline tasks/trigger/jenkins/0.1/.yaml:34]: jenkins-credentials-volume

File: tasks/trigger/jenkins/0.1/.yaml [hunk 7]
```diff
@@ -0,0 +1,117 @@
...
+          mountPath: /usr/local/konflux-test-infra
```
Comment (flacatus) [inline tasks/trigger/jenkins/0.1/.yaml:35]: /usr/local/jenkins-credentials-volume

**Review by flacatus — CHANGES_REQUESTED** [review]

File: docs/qe-available-tasks/trigger-jenkins-job.md [hunk 8]
```diff
@@ -0,0 +1,95 @@
...
+  username: username
+  apitoken: api-token
+  crumb: crumb
+```
+
+## Usage
+
+1. Without `JOB_PARAMS` parameters
+
+   ```yaml
```
Comment (flacatus) [inline docs/qe-available-tasks/trigger-jenkins-job.md:55]: I would add an example on how you can use it in a pipeline

File: tasks/trigger/jenkins/0.1/.yaml (hunk 6 above)
Comment (psturc) [reply tasks/trigger/jenkins/0.1/.yaml:34]: maybe that `volumeMount` is not needed at all? I see there are [secrets loaded in the env section already](https://github.com/konflux-ci/tekton-integration-catalog/pull/72/files#diff-402c5c867734253dc437be52558c523239d33bd7a65be2a15c5f15c2da905d7aR103)

**Review by flacatus — APPROVED** [review]

**Review by flacatus — APPROVED** [review]

File: docs/qe-available-tasks/trigger-jenkins-job.md [hunk 9]
```diff
@@ -0,0 +1,67 @@
+# Trigger Jenkins Job
+
+The following task can be used to trigger a Jenkins job using CURL request from a Tekton Task.
+
+More details on Remote Access API can be found [here](https://www.jenkins.io/doc/book/using/remote-access-api/)
+
+## Install the Task
+
+```bash
+kubectl apply -f https://api.hub.tekton.dev/v1/resource/tekton/task/trigger-jenkins-job/0.1/raw
```
Comment (psturc) [inline docs/qe-available-tasks/trigger-jenkins-job.md:10]: Do we need this?

File: docs/qe-available-tasks/trigger-jenkins-job.md [hunk 10]
```diff
@@ -0,0 +1,67 @@
...
+    - name: trigger-jenkins-job
+      taskRef:
+        resolver: "git"
+        params:
+        - name: url
+          value: https://github.com/konflux-ci/tekton-integration-catalog.git
+        - name: revision
+          value: main
+        - name: pathInRepo
+          value: tasks/trigger/jenkins/0.1/trigger-jenkins-job.yaml
```
Comment (psturc) [inline docs/qe-available-tasks/trigger-jenkins-job.md:61]:
```suggestion
          value: tasks/triggers/jenkins/0.1/trigger-jenkins-job.yaml
```

**Review by psturc — COMMENTED** [review]
just one question and one suggestion, otherwise lgtm

**Review by psturc — APPROVED** [review]

**Pull request merged**